include/qpid/dispatch/server.h

```diff
@@ -451,6 +452,28 @@ typedef struct qd_server_config_t {
      * connection will be automatically closed.
      */
     int idle_timeout_seconds;
+
+    /**
+     *  Holds comma separated list that indicates which components of the message should be logged.
+     *  Defaults to 'none' (log nothing). If you want all properties and application properties of the message logged use 'all'.
+     *  Specific components of the message can be logged by indicating the components via a comma separated list.
+     *  The components are
+     *  message-id
+     *   user-id
+     *   to
+     *   subject
+     *   reply-to
+     *   correlation-id
+     *   content-type
+     *   content-encoding
+     *   absolute-expiry-time
+     *   creation-time
+     *   group-id
+     *   group-sequence
+     *   reply-to-group-id
+     *   app-properties.
+     */
+    char *log_message;
```

Comment:
I would recommend storing this in its bit-mask form here.  Then it only needs to be parsed once at configuration time.  This also allows you to isolate the string definitions and the conversion functions to a single .h/.c module.